app/views/hiring_staff/vacancies/application_details/show.html.haml

```diff
@@ -1,6 +1,5 @@
 - content_for :page_title_prefix, page_title_prefix(@vacancy, @application_details_form, t('jobs.application_details'))
-= render 'hiring_staff/vacancies/vacancy_form_partials/heading'
-
+= render HiringStaff::VacancyFormPageHeadingComponent.new(@vacancy, session[:vacancy_attributes]&.[]('job_location'), session[:vacancy_attributes]&.[]('readable_job_location'))
```

Comment:
I would use a simpler interface:
```ruby
render HiringStaff::VacancyFormPageHeadingComponent.new(@vacancy, session[:vacancy_attributes])
```
and then have the logic to extract job_location and readable_job_location in the component

Reply:
Fair enough, it's a bit ugly to look at. I was trying to be specific about what arguments are needed in the component.